src/hooks/useTrades.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react'
+
+import { Trade } from 'api/exchange/ExchangeApi'
+
+import { getTrades } from 'services'
+
+import { appendTrades } from 'reducers-actions/trades'
+
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import useGlobalState from 'hooks/useGlobalState'
+import { CONFIRMATION_BLOCKS } from 'const'
+
+// TODO: (on global state) store trades to local storage
+// TODO: account for reverts
+export function useTrades(): Trade[] {
+  const [
+    {
+      trades: { trades, lastCheckedBlock },
+    },
+    dispatch,
+  ] = useGlobalState()
+  const { userAddress, networkId, blockNumber } = useWalletConnection()
+
+  useEffect(() => {
+    // Check only up to `latest - CONFIRMATION_BLOCKS` to reduce chance of re-orgs
+    const toBlock = blockNumber && blockNumber - CONFIRMATION_BLOCKS
+
+    if (userAddress && networkId && toBlock && (!lastCheckedBlock || lastCheckedBlock < toBlock)) {
+      getTrades({
+        userAddress,
+        networkId,
+        fromBlock: lastCheckedBlock,
```

Comment:
`fromBlock and toBlock` are inclusive boundaries
Try
```js
await exchangeApi.getPastTrades({networkId:1, fromBlock: 10154000, toBlock: 10154154})
await exchangeApi.getPastTrades({networkId:1, fromBlock: 10154154})
```
you'll get dulicate events in the second result

Have to do `fromBlock: lastCheckedBlock+1`

Reply:
This is the only comment that really needs to be addressed before merge